### PR TITLE
chore: Update test READMEs to reflect correct information on testing workflows

### PR DESCRIPTION
**Pull request by DharmitD**
**Description of your changes:**

- Remove obsolete information related to Prow tests, and gcloud dependency, since we've moved to GH Action Workflows/ Kind.
- Add details related to the newly added GHA workflows and how to replicate them locally/manually using Kind clusters.

Resolves https://github.com/kubeflow/pipelines/issues/11061

**Checklist:**
- [x] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -1,93 +1,100 @@
...
-
-## Troubleshooting
-
-**Q: Why is my test taking so long on GKE?**
-
-The cluster downloads a bunch of images during the first time the test runs. It will be faster the second time since the images are cached.
-The image building steps are running in parallel and usually takes 2~3 minutes in total. If you are experiencing high latency, it might due to the resource constrains
-on your GKE cluster. In that case you need to deploy a larger cluster. 
+1. **Install Prerequisites**:
+   - [Docker](https://docs.docker.com/get-docker/)
```
Comment (hbelmiro) [inline test/README.md:23]: Docker is not a prerequisite. For example, I use Podman :)
```suggestion
   - A container engine like [Podman](https://podman.io) or [Docker](https://docs.docker.com/get-docker/)
```

File: test/kfp-functional-test/README.md [hunk 2]
```diff
@@ -6,52 +6,71 @@
...
     ```bash
-    gcloud auth application-default login
+    cd {YOUR_ROOT_DIRECTORY_OF_KUBEFLOW_PIPELINES}
+    python3 ./test/kfp-functional-test/run_kfp_functional_test.py --host "http://localhost:8080"
     ```
-    acquire new user credentials to use for Application Default Credentials.
-    Credentials saved to file with {CREDENTIALS_PATH} similar to: [$HOME/.config/gcloud/application_default_credentials.json]
 
-1. copy the Credentials to the temp folder
+### Via Docker (Using Kind)
```
Comment (hbelmiro) [inline test/kfp-functional-test/README.md:44]:
```suggestion
### Via Kind
```

File: test/kfp-functional-test/README.md [hunk 3]
```diff
@@ -6,52 +6,71 @@
...
+    kubectl apply -k manifests/
+    ```
+
+3.  Ensure the cluster is ready:
+
+    ```bash
+    kubectl cluster-info --context kind-kfp-functional-test-cluster
+    ```
+
+4.  Start a Docker container and run the functional test:
```
Comment (hbelmiro) [inline test/kfp-functional-test/README.md:64]:
```suggestion
4.  Start a container and run the functional test:
```

**Review by hbelmiro — COMMENTED** [review]
Nice job @DharmitD. I left a few suggestions.

Also, can you please:

- Update/remove:
  - https://github.com/kubeflow/pipelines/blob/master/samples/test/README.md
  - https://github.com/kubeflow/pipelines/blob/master/sdk/python/test_data/README.md (I'm not sure that one is outdated)
- Link https://github.com/kubeflow/pipelines/issues/11061 to this PR

File: test/README.md (hunk 1 above)
Comment (DharmitD) [reply test/README.md]: Done, made it more inclusive!

File: test/kfp-functional-test/README.md (hunk 2 above)
Comment (DharmitD) [reply test/kfp-functional-test/README.md]: Done!

File: test/kfp-functional-test/README.md (hunk 3 above)
Comment (DharmitD) [reply test/kfp-functional-test/README.md]: Done.

**Comment by DharmitD** [comment]
> Update/remove:
> * https://github.com/kubeflow/pipelines/blob/master/samples/test/README.md
> * https://github.com/kubeflow/pipelines/blob/master/sdk/python/test_data/README.md (I'm not sure that one is outdated)

Thanks! I updated both these READMEs to point to the right files.

**Review by hbelmiro — APPROVED** [review]
/lgtm

**Comment by hbelmiro** [comment]
@chensun @HumairAK can you please approve if that makes sense to you?

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**